text/0000-introduce empty ranges.md

```diff
@@ -0,0 +1,267 @@
+In this RFC, this expression is used to define one portion of the now *empty* Ranges in accordance with the natural expectation and the implementation of Range in other languages. Using this test, `Range(0, 10, -1)` is *empty* and no longer *infinite*, but so is in fact `Range(0, inf, -1)`, even though any finite steps would mathematically never produce progress towards an infinite bound. While one can generally never produce a finite list of elements that would incrementally move from 0 to +Inf, we can know in the case of `Range(0, inf, -1)` that the list of elements must be empty because of the sign of the step parameter. 
+
+The other case in which a Range is now *empty* is when the bound identity `min == max` can be evaluated meaningfully even when `step` equals 0, or nan, or +-Inf. In the case of identical bounds, no iteration is needed regardless of the step magnitude. However, in order for the Range to be *empty*, this proposal requires the bounds to be finite, agreeing with the original implementation on the fact that one `inf` cannot be meaningfully tested for equality to another `inf`.
+
+The cases that would remain *infinite* under this proposal are therefore those which are: *not empty* AND which have either a `+Inf`, `-Inf`, or `NaN` values in any parameter or a zero `step` parameter.
+Here is the corresponding portion of the modified range.pony, illustrating these definitions:
+
+```pony
+class Range[A: (Real[A] val & Number) = USize] is Iterator[A]
+  ..
+  let _min: A
+  let _max: A
+  let _inc: A
+  let _forward: Bool
+  let _infinite: Bool
+  let _empty: Bool
+  var _idx: A
+
+  new create(min: A, max: A, inc: A = 1) =>
+    _min = min
+    _max = max
+    _inc = inc
+    _forward = (_min < _max) and (_inc > 0)
+    (let is_bounds_finite, let is_step_finite) =
+      iftype A <: FloatingPoint[A] then
+        (_min.finite() and _max.finite(), _inc.finite())
+      else
+        (true, true)
+      end
+    let no_progress = ((_min < _max) and (_inc < 0))
+                   or ((_min > _max) and (_inc > 0)) // false if any is NaN
+    _empty = no_progress
+          or ((_min == _max) and is_bounds_finite)
+    _infinite = not _empty
+            and ((_inc == 0)
+                or not is_bounds_finite
+                or not is_step_finite)
+    if _empty then _idx = _max else _idx = _min end // has_next() will return false without additional code
+```
+
+With a so-modified Range class and a new .is_emtpy() function, here is a list of Range examples all of which were previously infinite, and their new classification:
```

Comment:
It seems like everything below should also be included in the "how we test" section.